lasio/writer.py

```diff
@@ -110,7 +110,12 @@ def write(
     if STEP is None:
         if STOP != STRT:
             # prevents an error being thrown in the case of only a single sample being written
-            STEP = las.index[1] - las.index[0]  # Faster than np.gradient
+            # Example: 0.51 -> "51" -> len("51") -> 2
+            # So we'll make STEP with a fractional len of '2' after the decimal
+            fraction_len = len(str(las.index[1]).split(".")[1])
```

Comment:
does this work when the index curve is in integers?

Reply:
Good catch, I'll fix that for the case of integers and/or decimal mark is missing and post another commit.

Reply:
I've Added additional checks for fractional section of number.  If it does not exist, then fraction_len is defaulted to 0.

Also I added a check of the initial las.index[0] to make sure we get the largest fractional_len of the two numbers.

This approach is an attempt to retain the fractional len of the original data.  If this is an important goal, then the additional complexity may be worth it.  

I'll make a 2nd pull request that standardizes output on `fmt="%.5f"` for comparison.  To me, the `fmt="%.5f"` feels more reliable and maintainable. It has a sensible default and allows the user to pass in a different value for `fmt`.

Reply:
#432 the failing test have been updated to match `.5f` precision when writing the STEP value.  The tests pass now.